contracts/erc1155/ERC1155YungWkndBase.sol

```diff
@@ -0,0 +1,358 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.10;
+
+import "./ERC1155Base.sol";
+import "../metadata/MetadataEncryption.sol";
+import "../tokenManager/interfaces/IPostTransfer.sol";
+import "../tokenManager/interfaces/IPostBurn.sol";
+import "./interfaces/IHighlightRenderer.sol";
+import "../utils/ERC1155/ERC1155URIStorageUpgradeable.sol";
+import "./interfaces/IERC1155YungWkndMint.sol";
+
+/**
+ * @title Generalized Base ERC1155
+ * @author highlight.xyz
+ * @notice Generalized Base NFT smart contract
+ */
+abstract contract ERC1155YungWkndBase is ERC1155Base, ERC1155URIStorageUpgradeable, IERC1155YungWkndMint {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    uint256 _tokenCount;
+
+    /**
+     * @notice Throw when attempting to mint, while mint is frozen
+     */
+    error MintFrozen();
+
+    /**
+     * @notice Throw when requested token is not in range within bounds of limit supply
+     */
+    error TokenNotInRange();
+
+    /**
+     * @notice Throw when new supply is over limit supply
+     */
+    error OverLimitSupply();
+
+    /**
+     * @notice Throw when array lengths are mismatched
+     */
+    error MismatchedArrayLengths();
+
+    /**
+     * @notice Throw when string is empty
+     */
+    error EmptyString();
+
+    /**
+     * @notice Custom renderer config, used for collections where metadata is rendered "in-chain"
+     * @param renderer Renderer address
+     * @param processMintDataOnRenderer If true, process mint data on renderer
+     */
+    struct CustomRendererConfig {
+        address renderer;
+        bool processMintDataOnRenderer;
+    }
+
+    /**
+     * @notice Contract metadata
+     */
+    string public contractURI;
+
+    /**
+     * @notice Limit the supply to take advantage of over-promising in summation with multiple mint vectors
+     */
+    uint256 public limitSupply;
+
+    /**
+     * @notice Custom renderer config
+     */
+    CustomRendererConfig public customRendererConfig;
+
+    /**
+     * @notice Emitted when uris are set for tokens
+     * @param ids IDs of tokens to set uris for
+     * @param uris Uris to set on tokens
+     */
+    event TokenURIsSet(uint256[] ids, string[] uris);
+
+    /**
+     * @notice Emitted when limit supply is set
+     * @param newLimitSupply Limit supply to set
+     */
+    event LimitSupplySet(uint256 indexed newLimitSupply);
+
+    /**
+     * @notice See {IERC1155YungWkndMint-mintOneToOneRecipient}
+     */
+    function mintOneToOneRecipient(address recipient) external virtual onlyMinter nonReentrant returns (uint256) {
```

Comment:
The mint interface here (functions here and below) should be updated to accommodate 1155, so an additional `tokenID` parameter